tests/integration/targets/postgresql_privs/tasks/postgresql_privs_general.yml

```diff
@@ -653,6 +653,35 @@
     login_user: "{{ db_user3 }}"
     login_password: password
 
+# Issue https://github.com/ansible-collections/community.general/issues/994
+- name: Create a procedure for tests
+  postgresql_query:
+    query: "CREATE PROCEDURE mock_procedure() LANGUAGE SQL AS $$ SELECT 1; $$;"
+    db: "{{ db_name }}"
+    login_user: "{{ db_user3 }}"
+    login_password: password
+  when: postgres_version_resp.stdout is version('11', '>=')
+
+# Issue https://github.com/ansible-collections/community.general/issues/994
+- name: Try to run module against a procedure, not function
+  postgresql_privs:
+    type: function
+    state: present
+    privs: ALL
+    roles: "{{ db_user2 }}"
+    objs: ALL_IN_SCHEMA
```

Comment:
I never used postgres and this module but don't you think it would make it more understandable to use here explicitly the mock procedure name? At least one would see immediately the connection between the task before and this one.

Reply:
@martinm82 the issue (and the part of the code) relates to `ALL_IN_SCHEMA`. I created an issue to add support for `procedure` and then we could look at the topic more deeply (and this is when the object could help)